src/config.rs

```diff
@@ -0,0 +1,228 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::io::Write;
+use std::{fs, path};
+
+#[derive(Debug)]
+pub enum Error {
+    ConfigNotFound,
+    Io(std::io::Error),
+    InvalidConfig(ParseError),
+}
+
+#[derive(Debug)]
+pub enum ParseError {
+    // u32 is the line number where an error occured
+    UnknownSymbol(u32),
+    MissingCommand(u32),
+    CommandWithoutWhitespace(u32),
+}
+
+impl From<std::io::Error> for Error {
+    fn from(val: std::io::Error) -> Self {
+        if val.kind() == std::io::ErrorKind::NotFound {
+            Error::ConfigNotFound
+        } else {
+            Error::Io(val)
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Hotkey {
+    keysyms: Vec<evdev::Key>,
+    command: String,
+}
+
+impl Hotkey {
+    fn new(keysyms: Vec<evdev::Key>, command: String) -> Self {
+        Hotkey { keysyms, command }
+    }
+}
+
+pub fn load(path: path::PathBuf) -> Result<Vec<Hotkey>, Error> {
+    let file_contents = load_file_contents(path)?;
+    parse_contents(file_contents)
+}
+
+fn load_file_contents(path: path::PathBuf) -> Result<String, Error> {
+    let mut file = File::open(path)?;
+    let mut contents = String::new();
+    file.read_to_string(&mut contents)?;
+    Ok(contents)
+}
+
+fn parse_contents(contents: String) -> Result<Vec<Hotkey>, Error> {
+    let key_to_evdev_key: HashMap<&str, evdev::Key> = HashMap::from([
+        ("q", evdev::Key::KEY_Q),
```

Comment:
A lot of the repetition can be removed by importing `evdev::Key::*` and then just using `KEY_Q`...etc.
I don't think those identifiers will be used by anything else so It's probably fine to do that.